From b2f4eb1014ab115a4786421d3a85594e8e572655 Mon Sep 17 00:00:00 2001
From: icg_intg <icg_intg@intel.com>
Date: Thu, 10 May 2018 10:05:57 +0800
Subject: [PATCH 012/129] media: intel-ipu4: css: fw release against
 scci_IPU4_master_20180509_0030

Message for Open Source:
New firmware css release

Message for Internal:
[Issue/Feature]New firmware css release
[Root Cause/Changes]
isys & psys package lib2600b0 for commit-id 1aca3a6
FW_Manifest_ID: 657a6203c96c418282358404c66d2ccfdabac0de
FW_Build_Branch: IPU4_master
FW_Build_Topic: reduce_interrupts
library version: commit-id 1aca3a6
FW Bundle patch number of this release :  289155
FW tag : scci_IPU4_master_20180509_0030

Change-Id: Ia905df40255e609cd4a4dce04af9219c51db76f5
Depends-on: I0e4b1d289dd4cb5aa1c0447c98bc861040de1cb4
Depends-on: I375ac9cca5d34b26791c7c356abdddf2d9c9ae2e
Depends-on: If38519f8753b4e562f55cd223ee1c657140376df
Signed-off-by: icg_intg <icg_intg@intel.com>
Reviewed-on: https://icggerrit.corp.intel.com/327839
Reviewed-by: Coding Style
Approver: CaiyingX Geng <caiyingx.geng@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h       | 2 +-
 .../lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h        | 2 ++
 .../ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isysapi_types.h  | 2 ++
 .../intel/ipu4/ipu4-css/lib2600/isysapi/src/ia_css_isys_private.c   | 4 ++++
 .../lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h    | 2 +-
 .../psysapi/dynamic/interface/ia_css_psys_process_group_cmd_impl.h  | 6 ++++--
 6 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h
index c6f47ef..eff260d 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20180506
+#define IA_CSS_FW_PKG_RELEASE  0x20180509
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h
index aff4cfb..dc42fe5 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isys_fw_bridged_types.h
@@ -272,6 +272,8 @@ struct ia_css_isys_frame_buff_set_comm {
 	aligned_struct(struct ia_css_isys_param_pin_comm, process_group_light);
 	aligned_uint8(unsigned int, send_irq_sof);
 	aligned_uint8(unsigned int, send_irq_eof);
+	aligned_uint8(unsigned int, send_irq_capture_ack);
+	aligned_uint8(unsigned int, send_irq_capture_done);
 	aligned_uint8(unsigned int, send_resp_sof);
 	aligned_uint8(unsigned int, send_resp_eof);
 	aligned_uint8(unsigned int, frame_counter);
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isysapi_types.h b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isysapi_types.h
index cc00cb6..dafc345 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isysapi_types.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/interface/ia_css_isysapi_types.h
@@ -272,6 +272,8 @@ struct ia_css_isys_frame_buff_set {
 	struct ia_css_isys_param_pin process_group_light;
 	unsigned int send_irq_sof;
 	unsigned int send_irq_eof;
+	unsigned int send_irq_capture_ack;
+	unsigned int send_irq_capture_done;
 	unsigned int send_resp_sof;
 	unsigned int send_resp_eof;
 	uint8_t      frame_counter;
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/src/ia_css_isys_private.c b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/src/ia_css_isys_private.c
index f44a24e..ec92f14 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/src/ia_css_isys_private.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600/isysapi/src/ia_css_isys_private.c
@@ -601,6 +601,10 @@ STORAGE_CLASS_INLINE void frame_buff_set_host_to_css(
 			frame_buff_set_host->send_irq_sof ? 1 : 0;
 	frame_buff_set_css->send_irq_eof =
 			frame_buff_set_host->send_irq_eof ? 1 : 0;
+	frame_buff_set_css->send_irq_capture_done =
+			(uint8_t)frame_buff_set_host->send_irq_capture_done;
+	frame_buff_set_css->send_irq_capture_ack =
+			frame_buff_set_host->send_irq_capture_ack ? 1 : 0;
 	frame_buff_set_css->send_resp_sof =
 			frame_buff_set_host->send_irq_sof ?
 				1 : frame_buff_set_host->send_resp_sof;
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h
index c6f47ef..eff260d 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/DSS_V2_program_group/ia_css_fw_pkg_release.h
@@ -11,4 +11,4 @@
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 */
-#define IA_CSS_FW_PKG_RELEASE  0x20180506
+#define IA_CSS_FW_PKG_RELEASE  0x20180509
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/psysapi/dynamic/interface/ia_css_psys_process_group_cmd_impl.h b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/psysapi/dynamic/interface/ia_css_psys_process_group_cmd_impl.h
index f4b191e..530f93e 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/psysapi/dynamic/interface/ia_css_psys_process_group_cmd_impl.h
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/lib2600psys/lib/psysapi/dynamic/interface/ia_css_psys_process_group_cmd_impl.h
@@ -22,8 +22,8 @@
 #define N_UINT64_IN_PROCESS_GROUP_STRUCT	2
 #define N_UINT32_IN_PROCESS_GROUP_STRUCT	5
 #define N_UINT16_IN_PROCESS_GROUP_STRUCT	5
-#define N_UINT8_IN_PROCESS_GROUP_STRUCT		6
-#define N_PADDING_UINT8_IN_PROCESS_GROUP_STRUCT	4
+#define N_UINT8_IN_PROCESS_GROUP_STRUCT		7
+#define N_PADDING_UINT8_IN_PROCESS_GROUP_STRUCT	3
 
 #define SIZE_OF_PROCESS_GROUP_STRUCT_BITS \
 	(IA_CSS_RBM_BITS \
@@ -88,6 +88,8 @@ struct ia_css_process_group_s {
 	uint8_t base_queue_id;
 	/**< Number of dedicated queues used */
 	uint8_t num_queues;
+	/**< Mask the send_pg_done IRQ */
+	uint8_t mask_irq;
 	/**< Padding for 64bit alignment */
 	uint8_t padding[N_PADDING_UINT8_IN_PROCESS_GROUP_STRUCT];
 };
-- 
1.9.1

